process/interceptors/multiDataInterceptor.go

```diff
@@ -116,18 +111,21 @@ func (mdi *MultiDataInterceptor) ProcessReceivedMessage(message p2p.MessageP2P,
 
 		interceptedMultiData = append(interceptedMultiData, interceptedData)
 
-		if !mdi.dataVerifier.IsForCurrentShard(interceptedData) {
-			log.Trace("intercepted data is for other shards",
+		shouldProcess := interceptedData.IsForCurrentShard() || mdi.whiteListHandler.IsWhiteListed(interceptedData)
```

Comment:
maybe you can extract interceptedData.IsForCurrentShard() in one variable to not be called again in log.Trace below

Reply:
done